OPAM 1.2.1 & external solver

I was using OPAM (v1.2.0x86_64-Linux) & cudf_remote_proxy without any problem.
With the new version 1.2.1x86_64-Linux (and the same cudf) I have :

$ opam update
...
[WARNING] External solver failed:
          # opam-version 1.2.1
          # os           linux
          # command      cudf_remote_proxy
...
[ERROR] Please retry with option --use-internal-solver

Is there a specific setting to do ?

(also :
$ opam --solver=cudf_remote_proxy
doesn't work. What is the syntax ?)

--- a/src/client/opamState.ml
+++ b/src/client/opamState.ml
@@ -1436,6 +1436,10 @@ let load_config root =
         List.map (fun s -> s, None)
           [CString "packup"; CIdent "input"; CIdent "output";
            CString "-u"; CIdent "criteria"]
+      | [CString cmd, None] ->
+        (* Default args if the command is a single string *)
+        List.map (fun s -> s, None)
+          [CString cmd; CIdent "input"; CIdent "output"; CIdent "criteria"]
       | cmd -> cmd
     in
     fun ~input ~output ~criteria ->

--- a/src/solver/opamSolver.ml
+++ b/src/solver/opamSolver.ml
@@ -295,7 +295,10 @@ let resolve ?(verbose=true) universe ~orphans request =
         OpamCudf.to_actions add_orphan_packages u resp
       with Failure "opamSolver" ->
         OpamGlobals.error_and_exit
-          "Please retry with option --use-internal-solver"
+          "External solver failure, please fix your installation and check \
+           %s/config and variable $OPAMEXTERNALSOLVER.\n\
+           You may also retry with option --use-internal-solver"
+          !OpamGlobals.root_dir
     else OpamHeuristic.resolve ~verbose ~version_map add_orphan_packages u req in
   match resolve simple_universe cudf_request with
   | Conflicts _ as c -> c

